pkg/operator/staticpod/controller/staticpodfallback/static_pod_fallback_condition.go

```diff
@@ -49,18 +53,22 @@ func New(targetNamespace string,
 		ResyncEvery(6*time.Minute).
 		WithInformers(kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().Pods().Informer()).
 		ToController(
-			"StaticPodStateFallback", // don't change what is passed here unless you also remove the old FooDegraded condition
+			fd.controllerInstanceName,
 			eventRecorder,
 		), nil
 }
 
 // sync sets/unsets a StaticPodFallbackRevisionDegraded condition if a pod that matches the given label selector is annotated with FallbackForRevision
 func (fd *staticPodFallbackConditionController) sync(ctx context.Context, _ factory.SyncContext) (err error) {
-	degradedCondition := operatorv1.OperatorCondition{Type: "StaticPodFallbackRevisionDegraded", Status: operatorv1.ConditionFalse}
+	degradedCondition := &operatorv1.OperatorCondition{Type: "StaticPodFallbackRevisionDegraded", Status: operatorv1.ConditionFalse}
+	status := applyoperatorv1.OperatorStatus()
 	defer func() {
 		if err == nil {
-			if _, _, updateError := operatorv1helpers.UpdateStatus(ctx, fd.operatorClient, operatorv1helpers.UpdateConditionFn(degradedCondition)); updateError != nil {
-				err = updateError
+			status = status.WithConditions(
+				conditionToApplyOperatorCondition(degradedCondition),
```

Comment:
No.  Switch the conditions themselves to applyconfiguration types so we avoid accidents due to conversion.